pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/SegmentDeletionManager.java

```diff
@@ -193,4 +211,29 @@ protected void removeSegmentFromStore(String tableName, String segmentId) {
       LOGGER.info("localDiskDir is not configured, won't delete segment {} from disk", segmentId);
     }
   }
+
+  protected void removeSegmentsFromDeletedDirectory(int deleteRetention) {
```

Comment:
I may go the other direction, since this is very specific to deletion subdir which is 2 levels exactly.

for each dir {
  get files that are aged
  delete files
  if (dir is empty) {
   delete dir
  }
}

this way, we avoid getting dir twice. 

this could turn out to be important if we are on a remote FS (or even in NFS to minimize cross-network ops).